### PR TITLE
MSGPatterns options

### PR DESCRIPTION
**Pull request by jirkadanek**

### CODE REVIEW

**Comment by jirkadanek** [comment]
@rkubis This is now ready for review and merge.

File: lib/options/sender_option_parser.rb [hunk 1]
```diff
@@ -93,6 +95,20 @@ def initialize(args)
         @options.msg_content = msg_content
       end
 
+      # Message property
+      @opt_parser.on(
+        "-P",
+        "--msg-property KEY=VALUE",
+        String,
+        "property specified as KEY=VALUE (use '~' instead of '=' for auto-casting)"
+      ) do |msg_content_map_item|
```
Comment (rkubis) [inline lib/options/sender_option_parser.rb]: Rename msg_content_map_item to msg_property.

File: lib/options/sender_option_parser.rb [hunk 2]
```diff
@@ -93,6 +95,20 @@ def initialize(args)
...
+        if @options.msg_properties.nil?
+          @options.msg_properties = {}
+        end
+
+        key, value = parse_kv(msg_content_map_item)
```
Comment (rkubis) [inline lib/options/sender_option_parser.rb]: Rename msg_content_map_item to msg_property.

File: lib/options/sender_option_parser.rb [hunk 3]
```diff
@@ -221,6 +224,29 @@ def initialize(args)
 
       # Parse basic, common and specific options for sender client
       parse(args)
+    end
+
+    private
+    def parse_kv(msg_content_map_item)
```
Comment (rkubis) [inline lib/options/sender_option_parser.rb]: Rename argument to something unique and use it in whole function.

File: lib/options/sender_option_parser.rb [hunk 4]
```diff
@@ -99,6 +99,15 @@ def initialize(args)
         @options.msg_content = msg_content
       end
 
+      # Message content from file
+      @opt_parser.on(
+        "--msg-content-from-file FILE",
+        String,
+        "message content from file"
+      ) do |file|
+        @options.msg_content = File.read(file)
```
Comment (rkubis) [inline lib/options/sender_option_parser.rb:108]: Nice!

File: lib/options/sender_option_parser.rb (hunk 1 above)
Comment (jirkadanek) [reply lib/options/sender_option_parser.rb]: done

File: lib/options/sender_option_parser.rb (hunk 2 above)
Comment (jirkadanek) [reply lib/options/sender_option_parser.rb]: done

File: lib/options/sender_option_parser.rb (hunk 3 above)
Comment (jirkadanek) [reply lib/options/sender_option_parser.rb]: done

**Comment by rkubis** [comment]
Looks good, will merge.

**Pull request merged**